Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1

```diff
@@ -560,3 +561,296 @@ function UpdateSettingsFile {
+    - type: The type of the file (e.g., workflow, settings).
+    - destinationFullPath: The full path to the destination file in the target repository.
+#>
+function GetFilesToUpdate {
+    Param(
+        [Parameter(Mandatory=$true)]
+        $settings,
+        [Parameter(Mandatory=$true)]
+        $baseFolder,
+        [Parameter(Mandatory=$true)]
+        $templateFolder,
+        $originalTemplateFolder = $null,
+        $projects = @()
+    )
+
+    $filesToUpdate = GetDefaultFilesToUpdate -includeCustomTemplateFiles:$($null -ne $originalTemplateFolder)
+    $filesToUpdate += $settings.customALGoFiles.filesToUpdate
+    $filesToUpdate = ResolveFilePaths -sourceFolder $templateFolder -originalSourceFolder $originalTemplateFolder -destinationFolder $baseFolder -files $filesToUpdate -projects $projects
+
+    $filesToExclude = GetDefaultFilesToExclude -settings $settings
+    $filesToExclude += $settings.customALGoFiles.filesToExclude
+    $filesToExclude = ResolveFilePaths -sourceFolder $templateFolder -originalSourceFolder $originalTemplateFolder -destinationFolder $baseFolder -files $filesToExclude -projects $projects
+
+    # Exclude files from filesToExclude that are not in filesToUpdate
+    $filesToExclude = $filesToExclude | Where-Object {
+        $fileToExclude = $_
+        $include = $filesToUpdate | Where-Object { $_.sourceFullPath -eq $fileToExclude.sourceFullPath }
+        if(-not $include) {
+            Write-Host "Excluding file $($fileToExclude.sourceFullPath) from exclude list as it is not in the update list"
+        }
+        return $include
+    }
+
+    # Exclude files from filesToUpdate that are in filesToExclude
+    $filesToUpdate = $filesToUpdate | Where-Object {
+        $fileToUpdate = $_
+        $include = -not ($filesToExclude | Where-Object { $_.sourceFullPath -eq $fileToUpdate.sourceFullPath })
+        if(-not $include) {
+            Write-Host "Excluding file $($fileToUpdate.sourceFullPath) from update as it is in the exclude list"
+        }
+        return $include
+    }
+
+    # Apply unusedALGoSystemFiles logic
+    $unusedALGoSystemFiles = $settings.unusedALGoSystemFiles
+
+    # Exclude unusedALGoSystemFiles from $filesToUpdate and add them to $filesToExclude
+    $unusedFilesToExclude = $filesToUpdate | Where-Object { $unusedALGoSystemFiles -contains (Split-Path -Path $_.sourceFullPath -Leaf) }
+    if ($unusedFilesToExclude) {
+        Trace-DeprecationWarning "The 'unusedALGoSystemFiles' setting is deprecated and will be removed in future versions." -DeprecationTag "unusedALGoSystemFiles"
+
+        Write-Host "The following files are marked as unused and will be removed if they exist:"
+        $unusedFilesToExclude | ForEach-Object { Write-Host "- $($_.destinationFullPath)" }
+
+        $filesToUpdate = $filesToUpdate | Where-Object { $unusedALGoSystemFiles -notcontains (Split-Path -Path $_.sourceFullPath -Leaf) }
+        $filesToExclude += @($unusedFilesToExclude)
+    }
+

```

Comment:
Since above logic is quite complex, it could make sense to write a summary in the end.
Using Write-Host might be too verbose, but at least when troubleshooting it would be nice. :)
```
OutputDebug -message "File update summary:"
OutputDebug -message "- Files to update: $($filesToUpdate.Count)"
OutputDebug -message "- Files to exclude: $($filesToExclude.Count)"
$filesToUpdate | ForEach-Object { OutputDebug -message "  UPDATE: $($_.destinationFullPath)" }
$filesToExclude | ForEach-Object { OutputDebug -message "  EXCLUDE: $($_.destinationFullPath)" }
```